src/components/reservation/helper.ts

```diff
@@ -0,0 +1,17 @@
+import { AgencyBranch } from "../../core/fbs/model";
+
+export const getPreferredLocation = (id: string, array: AgencyBranch[]) => {
+  const locationItem = array.find((item) => item.branchId === id);
+  return locationItem ? locationItem.title : id;
+};
+
+export const getPreferredLocationText = (
+  preferred: string,
+  selected: string,
+  branches: AgencyBranch[]
+) =>
+  selected
+    ? getPreferredLocation(selected, branches)
+    : getPreferredLocation(preferred, branches);
```

Comment:
Consider what types are used for the arguments of this function.

The use of types here confused me quite a bit. We test for `selected` which is a string. This should only evaluate to `false` if the string is empty. However if the user does not have a preferred branch then I would expect `preferred` to be `null`. Perhaps this goes back to the API response and there is nothing we can do about this.

Reply:
eslint-disable-next-line react/jsx-props-no-spreading